include/stl2/detail/iterator/any_iterator.hpp

```diff
@@ -0,0 +1,230 @@
+                    };
+                }
+            }
+            return nullptr;
+        }
+
+        template <class RValueReference, InputIterator I>
+        iter_move_fn<RValueReference> exec_big(op o, blob *src, blob *dst) {
+            switch (o) {
+            case op::copy:
+                ++(dst->big = src->big)->cnt;
+               break;
+            case op::move:
+                dst->big = std::exchange(src->big, nullptr);
+                break;
+            case op::nuke:
+                if (0 == --src->big->cnt)
+                    delete static_cast<shared_iterator<I>*>(src->big);
+                break;
+            case op::bump:
+                ++static_cast<shared_iterator<I> *>(src->big)->it;
+                break;
+            case op::comp:
+                if (static_cast<shared_iterator<I> const *>(src->big)->it ==
+                    static_cast<shared_iterator<I> const *>(dst->big)->it) {
+                    // fallthrough
+            case op::rval:
+                    return +[](blob const &src) -> RValueReference {
+                        return iter_move(
+                            static_cast<shared_iterator<I> const *>(src.big)->it);
+                    };
+                }
+            }
+            return nullptr;
+        }
+
+        template<class Reference, class ValueType, class RValueReference>
+        struct cursor {
+        private:
+            blob data_ = { nullptr };
+            Reference (*deref_)(blob const &) =
+                &uninit_deref<Reference>;
+            iter_move_fn<RValueReference> (*exec_)(op, blob *, blob *) =
+                &uninit_noop<RValueReference>;
+
+            InputIterator{I} cursor(I i, small_tag) {
+                ::new (static_cast<void *>(&data_.tiny)) I(std::move(i));
+                deref_ = &deref_small<Reference, I>;
+                exec_ = &exec_small<RValueReference, I>;
+            }
+            InputIterator{I} cursor(I i, big_tag) {
+                data_.big = new shared_iterator<I>(std::move(i));
+                deref_ = &deref_big<Reference, I>;
+                exec_ = &exec_big<RValueReference, I>;
+            }
+        public:
+            using value_type = ValueType;
+            using single_pass = std::true_type;
+            struct mixin;
+            cursor() = default;
+            cursor(cursor &&that) {
+                that.exec_(op::move, &that.data_, &data_);
+                swap(deref_, that.deref_);
+                swap(exec_, that.exec_);
+            }
+            cursor(cursor const &that) {
+                that.exec_(op::copy, const_cast<blob *>(&that.data_), &data_);
+                deref_ = that.deref_;
+                exec_ = that.exec_;
+            }
+            cursor &operator=(cursor &&that) {
+                if (&that != this) {
+                    this->~cursor();
+                    ::new (static_cast<void *>(this))
+                        cursor(std::move(that));
```

Comment:
Sí, but I'm more concerned about performance, and I worry about small iterators that are no-throw in fact but not marked `noexcept`.